controllers/volumereplicationgroup_controller.go

```diff
@@ -199,38 +203,159 @@ func (v *VolumeReplicationGroupReconciler) Reconcile(ctx context.Context, req ct
 
 	log.Info("Processing VolumeReplicationGroup ", volRepGroup.Spec, " in ns: ", req.NamespacedName)
 
-	pvcList, err := v.HandlePersistentVolumeClaims(ctx, volRepGroup)
+	// examine the DeletionTimestamp to determine whether the object is under deletion
+	if !volRepGroup.GetDeletionTimestamp().IsZero() {
+		if err := v.runAndRemoveFinalizer(ctx, volRepGroup, vrgFinalizerName); err != nil {
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
So we are queuing without any logging.  Either log the reason why you are requeuing or `return ctrl.Result{}, err`

Reply:
ACK. Will log the reason.